### DISCUSSION TITLE
DistributionTool can't find Actions array from manifest.json

### DISCUSSION THREAD

**Discussion by CCheeses** [elgatosf/elgato#120] [Help] [answered]
The distribution tool gives the error:
`Exception when parsing the action from <file path here>\AppData\Local\Temp\com.ccheeses.webhooks.sdPlugin\manifest.json
Error: The plugin declares no action in the manifest.json file. Please add an "Actions" array containing some actions.`
However, the manifest.json contains:
>"Actions": [
		{
			"Icon": "actions/assets/action",
			"Name": "Webhook",
			"States": [
				{
					"Image": "actions/assets/action",
					"TitleAlignment": "middle",
					"FontSize": 10
				}
			],
			"Controllers": ["Keypad", "Encoder"],
			"Encoder": {
				"layout": "$A1",
                "Icon": "actions/assets/action.svg",
                "StackColor": "#0000FF",
                "TriggerDescription": {
                    "Rotate": "Unused",
                    "Push": "send http request",
                    "Touch": "send http request",
                    "LongTouch": "Unused"
                }
			},
			"supportedInMultiActions": true,
			"Tooltip": "This is an example tooltip",
			"UUID": "com.ccheeses.webhooks.action",
			"PropertyInspectorPath": "actions/property-inspector/inspector.html"
		

}

**Comment by ChekTek** [comment, marked as answer]
Hi @ComradeCheeses 

It looks like you are missing the action's `UUID`. The `UUID` should extend the plugins reverse domain name notation. 

For example if you plugins UUID is `com.ccheeses.webhooks`, then your action's UUID would be something like `com.ccheeses.webhooks.my-cool-action`. 

Stream Deck uses the `UUID` to identify where to send events. 

There may be some potential to improve the error messaging here to provide some more information from the CLI tool, so I'll take this feedback to our team.
